lib/src/base_request.dart

```diff
@@ -30,10 +30,10 @@ abstract class BaseRequest {
   ///
   /// This defaults to `null`, which indicates that the size of the request is
   /// not known in advance. May not be assigned a negative value.
-  int get contentLength => _contentLength;
-  int _contentLength;
+  int? get contentLength => _contentLength;
+  int? _contentLength;
 
-  set contentLength(int value) {
+  set contentLength(int? value) {
```

Comment:
This throws on null today, we should make it non-nullable instead.

We can use `ArgumentError.checkNotNull` to keep the throwing for legacy mode if we like.

Reply:
It doesn't throw on `null` – just if `<0`

Reply:
+1, this looks like it needs to accept `null`.

Reply:
Ah yes I misinterpreted it :D